--- a/sysdeps/unix/sysv/linux/powerpc/dl-sysdep.c
+++ b/sysdeps/unix/sysv/linux/powerpc/dl-sysdep.c
@@ -20,7 +20,7 @@
 #include <kernel-features.h>
 #include <ldsodefs.h>
 
-int __cache_line_size attribute_hidden;
+int __cache_line_size attribute_hidden __attribute__((weak));
 
 /* Scan the Aux Vector for the "Data Cache Block Size" entry.  If found
    verify that the static extern __cache_line_size is defined by checking
